### PR TITLE
Add support for line paint property functions 

### PR DESCRIPTION
**Pull request by lucaswoj**

### CODE REVIEW

**Comment by jfirebaugh** [comment]
[`circle.vertex.glsl` uses `uniform float u_devicepixelratio`](https://github.com/mapbox/mapbox-gl-shaders/blob/7971dc1705f20f453c3ed42f240f53ca39e31778/src/circle.vertex.glsl#L12). Any particular reasons to prefer a `#define DEVICE_PIXEL_RATIO` vs a uniform? Can we make it consistent?

**Comment by lucaswoj** [comment]
That looks like an oversight. The`#define` should be used in all cases.

**Comment by mollymerp** [comment]
@lucaswoj @jfirebaugh this is ready for review whenever you have a moment

File: src/line.fragment.glsl [hunk 1]
```diff
@@ -1,25 +1,28 @@
-uniform lowp float u_opacity;
-uniform float u_blur;
-
 #pragma mapbox: define lowp vec4 color
+#pragma mapbox: define lowp float blur
+#pragma mapbox: define lowp float opacity
+#pragma mapbox: define mediump float gapwidth
```
Comment (jfirebaugh) [inline src/line.fragment.glsl]: `gapwidth` doesn't appear to be used in the fragment shader -- I think we can remove.

**Comment by mollymerp** [comment]
good catch @jfirebaugh -- fixed

**Review by jfirebaugh — APPROVED** [review]
Good to merge as soon as both downstream PRs (https://github.com/mapbox/mapbox-gl-js/pull/3033 and https://github.com/mapbox/mapbox-gl-native/pull/6658) are green on the same shaders SHA.

**Pull request merged**